Userland/Libraries/LibCompress/Gzip.cpp

```diff
@@ -152,17 +152,17 @@ ErrorOr<Bytes> GzipDecompressor::read_some(Bytes bytes)
     return bytes.slice(0, total_read);
 }
 
-Optional<DeprecatedString> GzipDecompressor::describe_header(ReadonlyBytes bytes)
+ErrorOr<Optional<String>> GzipDecompressor::describe_header(ReadonlyBytes bytes)
 {
     if (bytes.size() < sizeof(BlockHeader))
-        return {};
+        return OptionalNone {};
 
     auto& header = *(reinterpret_cast<BlockHeader const*>(bytes.data()));
     if (!header.valid_magic_number() || !header.supported_by_implementation())
-        return {};
+        return OptionalNone {};
 
     LittleEndian<u32> original_size = *reinterpret_cast<u32 const*>(bytes.offset(bytes.size() - sizeof(u32)));
-    return DeprecatedString::formatted("last modified: {}, original size {}", Core::DateTime::from_timestamp(header.modification_time).to_deprecated_string(), (u32)original_size);
+    return TRY(String::formatted("last modified: {}, original size {}", TRY(Core::DateTime::from_timestamp(header.modification_time).to_string()), (u32)original_size));
```

Comment:
I think the `Core::DateTime` formatter should suffice,as we do not change the format, saving a temporary String creation